flows/parent_flow.py

```diff
@@ -1,47 +1,367 @@
-
+    client = get_client()
+    
+    # Hardcoded HPC type for now
+    hpc_type = "als"
+    prefect_logger.info(f"Starting job router (parent flow) for HPC: {hpc_type}")
+    
+    # Auto-select environment based on hpc_type
+    target_env = determine_best_environment(hpc_type)
+    prefect_logger.info(f"Selected target environment: {target_env}")
+    
+    # Execute each step in sequence based on the selected environment
     flow_run_id = ""
-    for params in params_list:
-        if flow_type == FlowType.podman:
-            flow_run_id = await launch_podman(
-                podman_params=params, prev_flow_run_id=flow_run_id
-            )
-        elif flow_type == FlowType.conda:
-            flow_run_id = await launch_conda(
-                conda_params=params, prev_flow_run_id=flow_run_id
-            )
-        elif flow_type == FlowType.slurm:
-            flow_run_id = await launch_slurm(
-                slurm_params=params, prev_flow_run_id=flow_run_id
-            )
-        elif flow_type == FlowType.docker:
-            flow_run_id = await launch_docker(
-                docker_params=params, prev_flow_run_id=flow_run_id
-            )
-        else:
-            prefect_logger.error("Flow type not supported")
-            raise ValueError("Flow type not supported")
-
-    pass
+    
+    for i, child_job_params in enumerate(params_list):
+        prefect_logger.info(f"Running step {i+1} of {len(params_list)}")
+        
+        try:
+            # Get model name and task
+            model_name = child_job_params.get("model_name", "")
+            task_name = child_job_params.get("task_name", "")
+            params = child_job_params.get("params", {})
+            
+            # Get algorithm details from MLflow
+            algorithm_details = get_algorithm_details_from_mlflow(model_name)
+            
+            # Get the appropriate python file name based on the task name
+            if task_name == "run":
+                python_file = algorithm_details.get("python_file", "")
+            elif task_name == "train":
+                python_file = algorithm_details.get("python_file_train", "")
+            elif task_name == "inference":
+                python_file = algorithm_details.get("python_file_inference", "")
+            elif task_name == "tune":
+                python_file = algorithm_details.get("python_file_tune", "")
+            else:
+                # For any other task, default to python_file
+                python_file = algorithm_details.get("python_file", "")
+            
+            if not python_file:
+                prefect_logger.error(f"No Python file found for task {task_name}")
+                raise ValueError(f"No Python file found for task {task_name}")
+            
+            if target_env == FlowType.conda:
+                # Prepare conda parameters
+                conda_relevant_params = {
+                    "conda_env_name": algorithm_details["conda_env"],
```

Comment:
If we have a custom `conda` worker, would this worker create the `conda` environment when a model is executed for the first time? If so, I wonder if we could track these environments using `Prefect Blocks`, `Redis`, or maybe something else that is relatively lightweight.
*Please note that these conda worker could be just a regular process worker that runs a bash script.